zdm/iteration.py

```diff
@@ -194,19 +215,41 @@ def update_grid(grid,pset):
 	alpha=pset[2]
 	gamma=pset[3]
 	sfrn=pset[4]
-	
+	H0value=pset[8]
+
 	# assumes all parameters following this are passed to DM distribution
 	smear_mean=pset[5]
 	smear_sigma=pset[6]
-	oldlEmin=np.log10(grid.Emin)
-	oldlEmax=np.log10(grid.Emax)
+
+	#taking all the old grid parameters
+    oldmask=grid.smear
+    oldsmeargrid=grid.smear_grid
+    
+	oldlEmin=np.log10(float(grid.Emin))
+	oldlEmax=np.log10(float(grid.Emax))
 	oldalpha=grid.alpha
 	oldgamma=grid.gamma
 	oldsfrn=grid.sfr_n
 	oldsmean=grid.smear_mean
 	oldssigma=grid.smear_sigma
+	oldH0value=grid.H0
 	new_sfr_smear=False #do not need to calculate new product of sfr and smearing
 	
+	##check if H0 value changed
+    
+    if oldH0value != pset[8]:
+        cos.set_cosmology(H0=H0value)
+        zDMgrid, zvals,dmvals,H0=misc_functions.get_zdm_grid(H0=H0value,new=True,plot=False,method='analytic')
```

Comment:
One issue with this implementation is that, for multiple grids, it will generate and load a new zDM grid. In theory however, this could be done and set constant for a range of grids. That is, this component of the zdm grid will be identical for ASKAP ICS and FE, and Parkes. Can you find a way to do this only once? Or alternately, outside this function, you can use the zdm grid from one 'grid' object and pass it to another one? In any case, see if you can find a speed-up.

Reply:
The `update_grid` function only works for a single grid right? Ex- in `get_likelihood` we are using a enumerate function to loop over every grid. And anyway creating the `zDMgrid, zvals, dmvals, H0` takes about **0.2 to 0.3 seconds** on my computer - shouldn't be too much of a problem! I could instead use the zdm grid from one grid object and pass to the other one. Let me know if I need to!